src/main/java/org/apache/sling/resourceresolver/impl/ResourceResolverFactoryActivator.java

```diff
@@ -79,6 +81,45 @@ private static final class FactoryRegistration {
         public volatile CommonResourceResolverFactoryImpl commonFactory;
     }
 
+    private static final class VanityPathConfigurer {
+        private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+        void configureVanityPathPrefixes(String[] pathPrefixes, String[] pathPrefixesFallback,
+                                         String pathPrefixesPropertyName, String pathPrefixesFallbackPropertyName,
+                                         Consumer<String[]> filteredPathPrefixesConsumer) {
+            if (pathPrefixes != null) {
+                configureVanityPathPrefixes(pathPrefixes, filteredPathPrefixesConsumer);
```

Comment:
Please log a WARN in the case where both the regular prefixes and the fallback ones are defined; this is probably a customer error. We should rely on the old behaviour in this case  and only take information from `config.resource_resolver_vanitypath_whitelist()`.